### PR TITLE
Fix math domain error in decaying poll rate

### LINKED ISSUE #77: Problem handling the decaying polling rate in the asynchronous execution policy
An error such as the following can appear:

```
./bin/reframe: unexpected error: math domain error
Traceback (most recent call last):
  File "/scratch/shared/leone/jenscscs/reframe-ci-leone-5d81168/reframe/frontend/cli.py", line 413, in main
    runner.runall(checks_matched, system)
  File "/scratch/shared/leone/jenscscs/reframe-ci-leone-5d81168/reframe/frontend/executors/__init__.py", line 176, in runall
    self._runall(checks, system)
  File "/scratch/shared/leone/jenscscs/reframe-ci-leone-5d81168/reframe/frontend/executors/__init__.py", line 246, in _runall
    self._policy.exit()
  File "/scratch/shared/leone/jenscscs/reframe-ci-leone-5d81168/reframe/frontend/executors/policies.py", line 288, in exit
    desired_rate = pollrate(t_elapsed, real_rate)
  File "/scratch/shared/leone/jenscscs/reframe-ci-leone-5d81168/reframe/frontend/executors/policies.py", line 88, in __call__
    self._init_poll_fn(init_rate)
  File "/scratch/shared/leone/jenscscs/reframe-ci-leone-5d81168/reframe/frontend/executors/policies.py", line 82, in _init_poll_fn
    self._c = math.log(self._a / (self._thres*self._b)) / self._decay
ValueError: math domain error
```
Problem can be reproduced on Leone.

Internal issue: https://madra.cscs.ch/scs/reframe/issues/580

### PR DESCRIPTION
**Pull request by teojgo**
* Check that `self._a` is not smaller that a minimum allowed value.

Fixes #77

### CODE REVIEW

File: reframe/frontend/executors/policies.py [hunk 1]
```diff
@@ -65,6 +65,8 @@ def run_check(self, check, partition, environ):
 
 
 class PollRateFunction:
+    _min_a = 0.01
```
Comment (victorusu) [inline reframe/frontend/executors/policies.py]: Does it make sense to have this variable defined in the `settings.py`?
Or Have it defined capitalized at the beginning of the file?

Comment (vkarak) [reply reframe/frontend/executors/policies.py]: @victorusu Generally, yes, but not in this issue. I will create another one.

Comment (teojgo) [reply reframe/frontend/executors/policies.py]: I added this one so that I do not enter a magic number in the middle of the code.

Comment (vkarak) [reply reframe/frontend/executors/policies.py]: @victorusu Another problem with exposing `_min_a` as-is is that it doesn't have a physical meaning. It is not the minimum poll rate.

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**